### PR TITLE
test: add socket error and remove reconnect from client

### PR DESCRIPTION
**Pull request by dnlup**
Don't error requests in client `onSocketError` on these codes:

* `UND_ERR_INFO`
* `UND_ERR_SOCKET`

Modify `client-reconnect.js` test. Add socket error test.

### CODE REVIEW

**Review by ronag — CHANGES_REQUESTED** [review]
Rather than removing the reconnect test. Please modify it

**Comment by dnlup** [comment]
I tried triggering a `UND_ERR_SOCKET` but I wasn't able to follow the same flow of that test. I'll try again, I am sure I missed something.

File: test/client-reconnect.js [hunk 1]
```diff
@@ -6,36 +6,42 @@ const { createServer } = require('http')
 const FakeTimers = require('@sinonjs/fake-timers')
```
Comment (dnlup) [inline test/client-reconnect.js:7]: I am not sure I got this right and completely understood how the reconnection works in this case.

Comment (dnlup) [reply test/client-reconnect.js:7]: I have seen similar tests in `client-request.js` if I am not mistaken.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ronag** [comment]
Thanks!

**Comment by dnlup** [comment]
Thank you @ronag for the help

**Pull request merged**